### PR TITLE
Add tests for all unary aten ops supported in eager mode

### PR DESCRIPTION
**Pull request by LironKesem**
**Description**: Describe your changes.
   Add tests for all uniary aten ops supported in eager mode
**Motivation and Context**
- Why is this change required? What problem does it solve?
   Support eager mode (see that it's working the ops)
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 1]
```diff
@@ -4,11 +4,49 @@
 # pylint: disable=missing-docstring
 
 import unittest
+from parameterized import parameterized
 
 import numpy as np
 import onnxruntime_pybind11_state as torch_ort
 import torch
 
+ops= [["abs",         torch.tensor([-1, -2, 3, -6, -7])],
```
Comment (jamill) [inline orttraining/orttraining/eager/test/ort_ops.py:13]: Would it be useful to include a comment describing what the structure of `ops` is? (e.g. what are the values of the tuple?)

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 2]
```diff
@@ -177,5 +215,40 @@ def test_argmax(self):
...
+
+        eval("torch."+test_name+"_(cpu_tensor)")
+        eval("torch."+test_name+"_(ort_tensor)")
+
+
+        assert torch.allclose(cpu_tensor, ort_tensor.cpu())
+
 if __name__ == "__main__":
-    unittest.main()
+    unittest.main(verbosity=2)
```
Comment (jamill) [inline orttraining/orttraining/eager/test/ort_ops.py:252]: Do we intend to pass in the verbosity here?

**Comment by WilBrady** [comment]
We'll need to fix the Eager Mode CI Pipelines. Looks like you need to add `parameterized` to the requirements.

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 3]
```diff
@@ -8,6 +8,50 @@
...
+# ["isinf",       ],
+# ["det"          ]]
+
+
+def rename_func_to_op(testcase_func, param_num, param):
+    return f"test_{parameterized.to_safe_name(str(param.args[0]))}"
+
+
+def rename_func_to_inplace(testcase_func, param_num, param):
+    return f"test_{parameterized.to_safe_name(str(param.args[0]))}_"
```
Comment (souptc) [inline orttraining/orttraining/eager/test/ort_ops.py:54]: thanks for working on this. we can think about generate the test cases from the generation script too.

**Review by souptc — APPROVED** [review]

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 4]
```diff
@@ -8,6 +8,55 @@
 import numpy as np
 import onnxruntime_pybind11_state as torch_ort
 import torch
+from parameterized import parameterized
+
+# OPS - is a list of  list of [test_operator, the tested_tensor]
+ops = [
+    ["abs", torch.tensor([-1, -2, 3, -6, -7])],
+    ["acos"],
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:16]: should this have the same limits as asin? Similar for atan?

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 5]
```diff
@@ -8,6 +8,55 @@
...
+    ["atanh", torch.rand(6).uniform_(-1, 1)],
+    ["asin", torch.rand(6).uniform_(-1, 1)],
+    ["atan"],
+    ["ceil"],
+    ["cos"],
+    ["cosh"],
+    ["erf"],
+    ["exp"],
+    ["floor"],
+    ["log", torch.abs(torch.rand(6)) + 0.5],
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:28]: a comment explaining the test input would be helpful, especially the + 0.5.

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 6]
```diff
@@ -8,6 +8,55 @@
...
+    ["reciprocal"],
+    ["neg", torch.randn(10)],
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:30]: why randn(10)?

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 7]
```diff
@@ -8,6 +8,55 @@
...
+    ["round"],
+    ["relu"],
+    ["sigmoid"],
+    ["sin"],
+    ["sinh"],
+    ["sqrt", torch.randn(10).uniform_(1, 9999)],
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:36]: looks like uniform_ will completely replace randn. Are you using randn to create a tensor of size 10? Also why use uniform_ for sqrt and not just rand to abs(rand) (the latter if nan is the issue)?

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 8]
```diff
@@ -8,6 +8,55 @@
...
+    ["tan"],
+    ["tanh"],
+]
+# the following unary ops not been tested:
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:40]: please add a couple comments about why these ops have not been tested. What is stopping us from testing them so the next person to look at this has the context.

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 9]
```diff
@@ -196,6 +245,50 @@ def test_argmax(self):
...
+        device = self.get_device()
+        cpu_tensor = tensor_test
+        ort_tensor = cpu_tensor.to(device)
+
+        cpu_out_tensor = torch.tensor([], dtype=tensor_test.dtype)
+        ort_out_tensor = cpu_out_tensor.to(device)
+
+        st_cpu = f"torch.{test_name}(cpu_tensor, out=cpu_out_tensor)"
+        st_ort = f"torch.{test_name}(ort_tensor, out=ort_out_tensor)"
+        cpu_result = eval(st_cpu)
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:285]: why are we not using `compile` here?

**Review by WilBrady — APPROVED** [review]

**Pull request merged**